TilingShader/Materials/tiled_background.script

```diff
@@ -37,6 +37,12 @@ end
 
 function update(self, dt)
 	self.offset.x = self.offset.x + (sprite_size.x /8 *dt) *self.speed.x
+	if math.abs(self.offset.x) >= 1 then
+		self.offset = {x = 0, y = self.offset.y}
```

Comment:
I would recommend that you try to avoid recreating the table in update if possible. It will generate quite a bit of garbage over time.